### DISCUSSION TITLE
Select payment card to use within cart

### DISCUSSION THREAD

**Discussion by VojtaSim** [swellstores/community#219] [Help] [answered]
I'd like to know if and how is possible to specify a payment card to use within cart. Given that account (customer) has multiple cards attached I'd like to show them in a list. Customer would then select the card to use for checkout. However no documentation mentions how to do that without entering the card details again through Stripe elements.

Also, how do I set a card as default (i.e. card on file) for the account?

**Comment by sags95** [comment]
Hey @VojtaSim , 

The default card is the one set as the `account.billing.account_card_id`

- To list saved cards, after authenticating the user you would call `await swell.account.listCards()`
- If you want to update the default card, you can then call:
```
await swell.account.update({
  billing: {
    account_card_id: '<new_account_card_id>'
  }
});
```

Details can be found here: https://developers.swell.is/frontend-api/credit-card-management

**Comment by VojtaSim** [comment]
So to select a card I want to use for current checkout I have to set it as default, or does you answer regards only about how to set the default card?

**Reply by sags95** [reply, marked as answer]
@VojtaSim to explicitly set a card for the checkout you can update the cart billing details by passing in the `card.token` of the card you want to use:

```
await swell.cart.update({
      billing: {
        card: {
          token: 'pm_***' //or card_***
        }
      }
    });
```

**Comment by ericingram** [comment]
You can also pass a credit card by ID:

```
  await swell.cart.update({
      billing: {
        account_card_id
      }
    }
  });
```

**Reply by VojtaSim** [reply]
Your approach did not work for me...

I updated cart's billing info like so:
```
"billing": {
    "account_card_id": "643....",
    "address1":"..",
    "address2":null,
    "city":"...",
    "state":null,
    "zip":"...",
    "country":"..."
}
```

and got response:
```
billing: {
    account_card_id: null
    address1: "..."
    // ...
}
```

then when the cart was submitted and converted to order, the order's billing info was this:
```
"billing": {
        "account_card_id": null,
        "address1": "...",
        "address2": null,
        "city": "...",
        "state": null,
        "zip": "...",
        "country": "..."
    },
```

and in admin it did not have the payment method attached.

<img width="1217" alt="Screenshot 2023-04-25 at 9 21 23" src="https://user-images.githubusercontent.com/4440970/234203684-91294ff4-ec83-4b5f-9a6c-4568b9570da7.png">

**Reply by ericingram** [reply]
Make sure you have `account_info_saved: true` set on the cart as well, if this is being done by API call. Also the account card ID has to match the account on the cart.
